packages/editor/src/plugins/image/controls/upload-button.tsx

```diff
@@ -55,11 +57,21 @@ export function UploadButton({
         </span>
         <input
           type="file"
+          multiple={config.onMultipleUploadCallback ? true : false}
           accept="image/*"
           className="sr-only"
           onChange={({ target }) => {
             if (target.files && target.files.length) {
-              void src.upload(target.files[0], config.upload)
+              const filesArray = Array.from(target.files)
+
+              // Upload the first file like normal
+              void src.upload(filesArray[0], config.upload)
+
+              // If multiple files are allowed and more than one file is selected,
+              // call the onMultipleUploadCallback callback with the remaining files
+              if (config.onMultipleUploadCallback && filesArray.length > 1) {
+                config.onMultipleUploadCallback(filesArray.slice(1))
```

Comment:
Todo: Find out how to insert new image plugins as children with these files assigned.